components/server/src/api/server.ts

```diff
@@ -294,6 +295,22 @@ export class API {
                     };
                     if (grpc_type === "unary" || grpc_type === "client_stream") {
                         return withRequestContext(async () => {
+                            const isCellDisabled = await getExperimentsClientForBackend().getValueAsync(
+                                "cell_disabled",
+                                false,
+                                {},
+                            );
+                            if (
+                                isCellDisabled &&
+                                requestContext.requestMethod === "gitpod.v1.UserService/getAuthenticatedUser"
```

Comment:
Why not put it into that method directly? :thinking:

Reply:
Good question. I think we don't currently have a way to have methods for unauthenticated users, so it would only work if the user logs in.

Reply:
Ah, forgot about that, you are right. :grimacing: 

The only other question I have: Shouldn't we also have that in the websocket implementation [here](https://github.com/gitpod-io/gitpod/blob/1963a8e6854f2bd48a0a32f2ea45af6df8c8b7f0/components/server/src/workspace/gitpod-server-impl.ts#L429-L431)?

Reply:
> Shouldn't we also have that in the websocket implementation [here](https://github.com/gitpod-io/gitpod/blob/1963a8e6854f2bd48a0a32f2ea45af6df8c8b7f0/components/server/src/workspace/gitpod-server-impl.ts#L429-L431)?

I took a look at our feature flag situation and it looks like we actually have `dashboard_public_api_user_enabled` enabled on all environments, meaning that adding it to the websocket path most likely isn't necessary. I'm happy to add it, but it looks fine to leave out for now

Reply:
Added checks for websockets in [`4ab8aec` (#20222)](https://github.com/gitpod-io/gitpod/pull/20222/commits/4ab8aecd42bf5fa0f3859f66927ade1ac4f7d479)

Reply:
Thank you @filiptronicek ! Let's go!